src/main/scala/io/iohk/ethereum/network/BlockBroadcastActor.scala

```diff
@@ -0,0 +1,172 @@
+package io.iohk.ethereum.network
+
+import akka.actor.SupervisorStrategy.Stop
+import akka.actor.{Actor, ActorLogging, ActorRef, OneForOneStrategy, Props}
+import akka.agent.Agent
+import akka.util.ByteString
+import io.iohk.ethereum.domain.{Block, BlockHeader, Blockchain}
+import io.iohk.ethereum.network.PeerActor.MessageReceived
+import io.iohk.ethereum.network.PeerManagerActor.{GetPeers, Peer}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+import io.iohk.ethereum.network.p2p.messages.PV62
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.network.p2p.validators.BlockHeaderError.HeaderParentNotFoundError
+import io.iohk.ethereum.network.p2p.validators.{BlockHeaderValidator, BlockValidator}
+import io.iohk.ethereum.utils.{BlockchainStatus, NodeStatus}
+import org.spongycastle.util.encoders.Hex
+
+//FIXME: Handle peers not responding to block headers and bodies request [EC-107]
+class BlockBroadcastActor(
+  nodeStatusHolder: Agent[NodeStatus],
+  peer: ActorRef,
+  peerManagerActor: ActorRef,
+  blockchain: Blockchain) extends Actor with ActorLogging {
+
+  import BlockBroadcastActor._
+
+  override val supervisorStrategy =
+    OneForOneStrategy() {
+      case _ => Stop
+    }
+
+  override def receive: Receive = idle
+
+  private val msgsToSubscribe = Set(NewBlock.code, PV62.NewBlockHashes.code, BlockHeaders.code, BlockBodies.code)
+
+  def idle: Receive = {
+    case StartBlockBroadcast =>
+      peer ! PeerActor.Subscribe(msgsToSubscribe)
+      context become processMessages(ProcessingState(Nil, Nil, Nil, Nil))
+  }
+
+  def processMessages(state: ProcessingState): Receive =
+    handleReceivedMessages(state) orElse {
+
+      case ProcessNewBlocks if state.unprocessedBlocks.nonEmpty =>
+        if(state.unprocessedBlocks.tail.nonEmpty) self ! ProcessNewBlocks
+
+        val blockToProcess = state.unprocessedBlocks.head
+        val blockHeader = BlockHeaderValidator.validate(blockToProcess.header, blockchain)
+        val blockParentTotalDifficulty = blockchain.getTotalDifficultyByHash(blockToProcess.header.parentHash)
+
+        (blockHeader, blockParentTotalDifficulty) match {
+          case (Right(_), Some(parentTD)) =>
+            val blockToProcessDifficulty = parentTD + blockToProcess.header.difficulty
+            importBlockToBlockchain(blockToProcess, blockToProcessDifficulty)
+
+            peerManagerActor ! GetPeers
+            val newState = state.copy(
+              unprocessedBlocks = state.unprocessedBlocks.tail,
+              toBroadcastBlocks = state.toBroadcastBlocks :+ NewBlock(blockToProcess, blockToProcessDifficulty)
```

Comment:
as I see we do not broadcast `toBroadcastBlocks`, this is intended because it will be handle in different ticket?